app/models/bike.rb

```diff
@@ -497,6 +497,27 @@ def external_image_urls
     b_params.map { |bp| bp.external_image_urls }.flatten.reject(&:blank?).uniq
   end
 
+  def alert_image_url
+    return if current_stolen_record.blank?
+
+    generate_alert_image
+    current_stolen_record.alert_image_url
+  end
+
+  def generate_alert_image
+    return if current_stolen_record.blank?
+    return if current_stolen_record.alert_image.present?
+
+    # Remove any old alert images
+    stolen_records.each { |sr| sr.alert_image&.remove! }
```

Comment:
Wouldn't we want to trigger this when `current_stolen_record.blank?` too?

Reply:
Or maybe, I don't know that this fits will trigger all the times we want it to trigger. Would it make more sense on an after commit hook on stolen record?

Reply:
> Wouldn't we want to also trigger this when current_stolen_record.blank?

I would think not? If the given bike isn't currently stolen, then it shouldn't have an alert image (the template says "stolen bike alert"). I might be imagining the invocation sequence differently, though.

> Would it make more sense on an after commit hook on stolen record?

Yeah, I think so... I had some concern about triggering the callback excessively in an after_commit hook, but it'll also reduce storage requirements.

With an `after_create`, a newly created stolen record will trigger the creation of an alert image (if there's a bike image available). When a bike is recovered, the alert image sticks around until a new one is needed.

With an `after_commit`, the outdated alert image is removed when the bike is recovered.

Reply:
I don't know that the creation needs to be in an `after_` hook - I think the triggering based on the call to the alert image url makes sense and is slick.

I was just thinking specifically about the deletion. e.g. anytime the record is switched to be recovered, the alert image should deleted. Rather then delete method here, where it seems like it won't consistently delete the image.

Reply:
Updated -- moved image removal to an `after_commit` hook
```rb
# app/models/bike.rb L530-534 (55c09ed7)

def alert_image_url
  return if current_stolen_record.blank? || public_images.none?
  current_stolen_record.generate_alert_image
  current_stolen_record.alert_image_url
end
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/55c09ed7/app/models/bike.rb#L530-L534)]</sup>


```rb
# app/models/stolen_record.rb L53-55 (a0dc6a32)

mount_uploader :alert_image, AlertImageUploader
process_in_background :alert_image
after_commit :remove_outdated_alert_image
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/a0dc6a32/app/models/stolen_record.rb#L53-L55)]</sup>

```rb
# app/models/stolen_record.rb L283-286 (a0dc6a32)

# If the bike has been recovered, remove the alert_image
def remove_outdated_alert_image
  alert_image.remove! if date_recovered.present?
end
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/a0dc6a32/app/models/stolen_record.rb#L283-L286)]</sup>